incentivization.md

```diff
@@ -0,0 +1,240 @@
+An incentive scheme links the payoffs to whether nodes follow the protocol or not.
+Good behavior should be rewarded, bad behavior punished.
+
+An incentive scheme should balance the costs and benefits for a node.
+Rewards should compensate the cost of good behavior.
+Punishments should offset the benefits that bad behavior may bring.
+
+Let us analyze the costs and benefit of a server that are specific to the Store protocol:
+- storage;
+- bandwidth;
+- computation.
+
+Let us assume a constant flow of messages per epoch and a constant flow of requests for older messages.
+There are two processes: storing incoming messages, and serving old messages to clients.
+
+The cost of storing incoming messages for one epoch is composed of:
+- storage:
+	- storage costs of all older messages: proportional to cumulative (message size x time stored);
+	- storage costs of newly arrived messages: proportional to message size;
+	- a constant cost for I/O operations (storing new messages);
+- bandwidth (download) for receiving new messages: proportional to the total size of incoming messages per epoch;
+- computational costs of receiving and storing new messages.
+
+(Strictly speaking, the I/O cost may not always be constant due to caching, disk fragmentation, etc.)
+
+The cost of storing messages to clients, per epoch, is composed of:
+- storage: none (it's accounted for as storing cost);
+- bandwidth
+	- upload: proportional to (number of clients) x (length of time frame requested) x (message size);
+	- download: proportional to the number of requests;
+- computational cost of handling requests.
+
+TODO: write this down mathematically.
+
+Storage is likely the dominating cost.
+Storage costs is proportional to the amount of information stored and the time it is stored for.
+A cumulative cost of storing a single message grows linearly with time.
+Assuming a constant stream of new messages, the total storage cost is quadratic in time.
+
+The number of messages in a response may be approximated by the length of the time frame requested.
+This assumes that messages are broadcast in the Relay network at a constant rate.
+
+Computation: the server spends computing cycles while handling requests.
+This costs likely depends not only on the computation itself, but also at the database structure.
+For example, retrieving old or rarely requested messages from the local database may be more expensive than fresh or popular ones due to caching.
+
+TODO: In file storage, I store a file and I pay for the ability to query it later. In Store, Alice relays a message, a server stores is, and later Bob queries it (and pays for it under an i13n scheme). Is there a mismatch between who incurs costs and who pays for it? Shall we think of ways to make Alice incur some costs too? See: https://github.com/waku-org/research/issues/32
+
+# Incentivization tools
+
+We can think of incentivization tools as a two-by-two matrix:
```

Comment:
really like this framing. useful for the whitepaper for sure. perhaps we can add an image/table with it?

Reply:
its a great framework to asses protocols. eg:
* relay: x ticked in reputation reward+punishment
* rln-relay: x ticked in monetary punishemnt (if we enable slashing)
* light clients using store: x ticket in monetary reward?